Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -3395,6 +3396,26 @@ void WebPage::mouseEvent(const WebMouseEvent& mouseEvent, std::optional<Vector<S
     revokeSandboxExtensions(mouseEventSandboxExtensions);
 }
 
+void WebPage::performHitTestAndGetUserData(const WebMouseEvent& event, CompletionHandler<void(WebHitTestResultData&&, OptionSet<WebEventModifier>&&, UserData&&)>&& completionHandler)
+{
+    auto* localMainFrame = dynamicDowncast<WebCore::LocalFrame>(corePage()->mainFrame());
```

Comment:
Nit - `RefPtr` here, to guarantee that the pointer remains valid over its entire scope.

Reply:
Good idea -- fixed in https://github.com/WebKit/WebKit/pull/14221/commits/c88025563e2436195514cb0c0a22883be08bf627